gemrb/plugins/CREImporter/CREImporter.cpp

```diff
@@ -1299,85 +1243,50 @@ Effect *CREImporter::GetEffect()
-	act->BaseStats[IE_PICKPOCKET]=tmpByte;
-	str->Read( &tmpByte, 1 );
-	act->BaseStats[IE_FATIGUE]=tmpByte;
-	str->Read( &tmpByte, 1 );
-	act->BaseStats[IE_INTOXICATION]=tmpByte;
-	str->Read( &tmpByte, 1 );
-	act->BaseStats[IE_LUCK]=(ieByteSigned) tmpByte;
-	str->Read( &tmpByte, 1 );
+	str->Seek(2, GEM_CURRENT_POS);
+	ieWordSigned tmpWord;
+	str->ReadScalar(tmpWord);
+	act->AC.SetNatural(tmpWord);
+	str->ReadScalar<Actor::stat_t, ieWordSigned>(act->BaseStats[IE_ACCRUSHINGMOD]);
+	str->ReadScalar<Actor::stat_t, ieWordSigned>(act->BaseStats[IE_ACMISSILEMOD]);
+	str->ReadScalar<Actor::stat_t, ieWordSigned>(act->BaseStats[IE_ACPIERCINGMOD]);
+	str->ReadScalar<Actor::stat_t, ieWordSigned>(act->BaseStats[IE_ACSLASHINGMOD]);
+	ieByteSigned tmpByte;
+	str->ReadScalar(tmpByte);
+	act->ToHit.SetBase(tmpByte);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_NUMBEROFATTACKS]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_SAVEVSDEATH]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_SAVEVSWANDS]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_SAVEVSPOLY]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_SAVEVSBREATH]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_SAVEVSSPELL]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTFIRE]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTCOLD]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTELECTRICITY]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTACID]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTMAGIC]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTMAGICFIRE]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTMAGICCOLD]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTSLASHING]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTCRUSHING]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTPIERCING]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_RESISTMISSILE]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_DETECTILLUSIONS]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_SETTRAPS]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_LORE]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_LOCKPICKING]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_STEALTH]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_TRAPS]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_PICKPOCKET]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_FATIGUE]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_INTOXICATION]);
+	str->ReadScalar<Actor::stat_t, ieByte>(act->BaseStats[IE_LUCK]);
+	str->ReadScalar<Actor::stat_t, ieByteSigned>(act->BaseStats[IE_TRACKING]);
```

Comment:
luck should be signed; tracking wasn't, but this is a good change.

Reply:
you mean both tracking and luck should be signed?

Reply:
You inverted the previous way they were treated, hence the comment. It doesn't really matter for tracking.

Reply:
I can see that, but you said it is a good change, so I was asking for clarity on what that was to mean.

I'll change them around to match the original code. If we should change signedness of anything we can address that later.